New Kestrel Analytics to Detect Lateral Movement

leila-rashidi opened the pull request:
Hi, 

My colleague and I have developed a new Kestrel analytics to detect lateral movement using graph learning.

leila-rashidi committed:
Add files via upload

leila-rashidi committed:
Delete Dockerfile

leila-rashidi committed:
Delete Readme.md

leila-rashidi committed:
Delete classify.py

leila-rashidi committed:
Delete deepwalk.py

leila-rashidi committed:
Delete main.py

leila-rashidi committed:
Delete walker.py

leila-rashidi committed:
Create Readme.md

leila-rashidi committed:
Create main.py

leila-rashidi committed:
Add files via upload

subbyte reviewed (commented):
Thank you @leila-rashidi for the graph-based LM detector!

A few tasks to formalize it:
1. Could you add a description to README about the mechanism of this analytics and how it is different from the last LM analytics you did? So users get a better idea which one to use in different cases.
2. Could you state in the README whether this can be called via a Python analytics interface, or only Docker interface?
3. Could you put a test case (with STIX bundle) for this new analytics? Please refer to [my other comment](https://github.com/opencybersecurityalliance/kestrel-analytics/pull/16#issuecomment-1507338130) for more information.

leila-rashidi committed:
Add files via upload

Merged all previous commits, and changed the ReadMe as requested by Dr. Shu. Also, I added an SQL table for train and test ML approach.

leila-rashidi committed:
Add files via upload

Applying comments of Dr. Shu.

leila-rashidi committed:
Merge pull request #1 from leila-rashidi/leila-rashidi-patch-1

Add files via upload

leila-rashidi committed:
Add files via upload

A new commits after applying all comments of Dr. Shu

leila-rashidi committed:
Merge pull request #2 from leila-rashidi/leila-rashidi-patch-2

Add files via upload

leila-rashidi committed:
Merge branch 'opencybersecurityalliance:release' into release
